libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -818,7 +818,9 @@ private void cleanUp(Activity frontActivity, Account account) {
 
         // Finishes front activity if specified, and if this is the last account.
         if (frontActivity != null && (users == null || users.size() <= 1)) {
-            frontActivity.finish();
+            // only finish if it isn't hybrid or if hybrid, requires authentication
+            if (!isHybrid() || BootConfig.getBootConfig(frontActivity).shouldAuthenticate())
```

Comment:
@smcnulty-sfdc We do want to finish the hybrid activity in our hybrid apps though. Could the caller not pass in `frontActivity` instead? Or start it up again post-logout?

Reply:
Can we move this to `SalesforceDroidGapActivity` instead to pass in a `null` value for `frontActivity` when `logout` is called in the deferred auth context?